### PR TITLE
Observe Ingress AppsDomain For Routing Domain Override

### PR DESCRIPTION
**Pull request by dustman9000**
This is related to https://github.com/openshift/enhancements/pull/432 and [RFE-952](https://issues.redhat.com/browse/RFE-952)

This checks for the existence of Ingress AppsDomain to customize the routing domain.

Tested with a local build on a 4.6 OSD cluster:
```
$ oc edit ingresses.config/cluster
$ oc get ingresses.config/cluster -o json | jq .spec
{
  "appsDomain": "apps.fidata.io",
  "domain": "apps.drow-10-20-20.x2m8.s1.devshift.org"
}
$ oc expose service hello-openshift
route.route.openshift.io/hello-openshift exposed
$ oc get routes
NAME              HOST/PORT                                   PATH   SERVICES          PORT       TERMINATION   WILDCARD
hello-openshift   hello-openshift-my-project.apps.fidata.io          hello-openshift   8080-tcp                 None
```

### CODE REVIEW

File: pkg/operator/configobservation/ingresses/observe_ingresses.go [hunk 1]
```diff
@@ -40,6 +40,14 @@ func ObserveIngressDomain(genericListers configobserver.Listers, recorder events
 	}
 
 	routingDomain := configIngress.Spec.Domain
+	appsDomain := configIngress.Spec.AppsDomain
+
+	klog.Infof("routingDomain: %s, appsDomain: %s", routingDomain, appsDomain)
```
Comment (dustman9000) [inline pkg/operator/configobservation/ingresses/observe_ingresses.go:45]: This is debug that will be removed.

**Comment by Miciah** [comment]
Thanks!
/lgtm

**Comment by Miciah** [comment]
/retitle Bug 1891952: Observe Ingress AppsDomain For Routing Domain Override

**Pull request merged**